server/src/infra/repositories/job.repository.ts

```diff
@@ -1,23 +1,80 @@
-import { IJobRepository, JobCounts, JobItem, JobName, JOBS_TO_QUEUE, QueueName, QueueStatus } from '@app/domain';
+import {
+  IJobRepository,
+  JobCounts,
+  JobItem,
+  JobName,
+  JOBS_TO_QUEUE,
+  QueueName,
+  QueueStatus,
+  validateCronExpression,
+} from '@app/domain';
 import { getQueueToken } from '@nestjs/bullmq';
 import { Injectable, Logger } from '@nestjs/common';
 import { ModuleRef } from '@nestjs/core';
+import { SchedulerRegistry } from '@nestjs/schedule';
 import { Job, JobsOptions, Processor, Queue, Worker, WorkerOptions } from 'bullmq';
+import { CronJob, CronTime } from 'cron';
 import { bullConfig } from '../infra.config';
 
 @Injectable()
 export class JobRepository implements IJobRepository {
   private workers: Partial<Record<QueueName, Worker>> = {};
   private logger = new Logger(JobRepository.name);
 
-  constructor(private moduleRef: ModuleRef) {}
+  constructor(
+    private moduleRef: ModuleRef,
+    private schedulerReqistry: SchedulerRegistry,
+  ) {}
 
   addHandler(queueName: QueueName, concurrency: number, handler: (item: JobItem) => Promise<void>) {
     const workerHandler: Processor = async (job: Job) => handler(job as JobItem);
     const workerOptions: WorkerOptions = { ...bullConfig, concurrency };
     this.workers[queueName] = new Worker(queueName, workerHandler, workerOptions);
   }
 
+  validateCronExpression(expression: string) {
```

Comment:
If we are using the class-validator why is this needed still?

Reply:
Do validate e.g. config files

Reply:
To validate e.g. config files and also just to make really sure that the expression is valid because otherwise the system will die quite badly.